tests/VersionRouteTest.php

```diff
@@ -289,4 +291,61 @@ public function testVersion(string $version, bool $isCorrectVersion, string $rou
 
         $response->assertStatus($status);
     }
+
+    public function testWithoutApiVersion(): void
+    {
+        $testCaseMock = (new TestCaseMock('name'))
+            ->setUpMock($this->app)
+            ->withoutAPIVersion();
+
+        Route::get('/test', function () {
+            return 'test';
+        });
+
+        $response = $testCaseMock->json('get', '/test');
+
+        $response->assertOk();
+    }
+
+    public function testRouteWithSetApiVersion(): void
+    {
+        $version = $this->createMock(VersionEnumContract::class);
+        $version->value = VersionEnum::v1;
+
+        $testCaseMock = (new TestCaseMock('name'))
+            ->setUpMock($this->app)
+            ->setAPIVersion($version);
+
+        Route::version($version)->group(function () {
+            Route::get('/test', function () {
+                return 'test';
+            });
+        });
+
+        $response = $testCaseMock->json('get', '/test/');
+
+        $response->assertOk();
+    }
+
+    public function testRouteWithIncorrectVersion(): void
+    {
+        $version = $this->createMock(VersionEnumContract::class);
+        $version->value = VersionEnum::v1;
```

Comment:
let's change VersionEnum to the normal enum structure as package require php 8.3